packages-ts/gauntlet-terra-contracts/src/commands/tooling/upload.ts

```diff
@@ -42,20 +43,30 @@ export default class UploadContractCode extends TerraCommand {
 
     const contractReceipts = {}
     const responses: any[] = []
+    const parsedRetryCount = parseInt(this.flags.maxRetry)
+    const maxRetry = parsedRetryCount ? parsedRetryCount : 5
     for (let contractName of askedContracts) {
+      await prompt(`Uploading contract ${contractName}, do you wish to continue?`)
       const contract = await getContract(contractName, this.flags.version)
       console.log('CONTRACT Bytecode exists:', !!contract.bytecode)
-      try {
-        const res = await this.upload(contract.bytecode, contractName)
+      for (let retry = 0; retry < maxRetry; retry++) {
+        try {
+          const res = await this.upload(contract.bytecode, contractName)
 
-        logger.success(`Contract ${contractName} code uploaded succesfully`)
-        contractReceipts[contractName] = res.tx
-        responses.push({
-          tx: res,
-          contract: null,
-        })
-      } catch (e) {
-        logger.error(`Error deploying ${contractName} code: ${e.message}`)
+          logger.success(`Contract ${contractName} code uploaded succesfully`)
+          contractReceipts[contractName] = res.tx
+          responses.push({
+            tx: res,
+            contract: null,
+          })
+        } catch (e) {
+          logger.error(`Error deploying ${contractName} on attempt ${retry + 1} with the error: ${e.message}`)
+          if (maxRetry === retry + 1) {
+            throw e
```

Comment:
Can we rethrow it like `throw new Error(e.response.data)`? Try to safely access the `data`